frontend/src/api/collimator/models/class-properties.ts

```diff
@@ -3,4 +3,7 @@ type NonFunctionPropertyNames<T> = {
   [K in keyof T]: T[K] extends Function ? never : K;
 }[keyof T];
 
-export type ClassProperties<T> = Pick<T, NonFunctionPropertyNames<T>>;
+export type ClassProperties<T, TOmit extends string = ""> = Omit<
+  Pick<T, NonFunctionPropertyNames<T>>,
+  TOmit
+>;
```

Comment:
I haven't seen where this is used (swamped by the amount of changes), but it seems that adding an `Omit<,>` where needed would be more explicit and clear.